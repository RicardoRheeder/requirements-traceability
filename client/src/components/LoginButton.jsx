import React from "react";

import { setLoggedIn } from "../redux/stores/common/actions";
import { useDispatch } from "react-redux";
import { useAuth0 } from "@auth0/auth0-react";

export default function LoginButton() {
  const dispatch = useDispatch();
  const { loginWithRedirect } = useAuth0();

  return (
    <button
      id="LogInButton"
      className="login-button"
      onClick={() => {
        // dispatch(setLoggedIn(true));
        loginWithRedirect();
      }}
    >
      Log in
    </button>
  );
}
