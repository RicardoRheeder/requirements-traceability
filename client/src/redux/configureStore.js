import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

// restricting extension in production using process.env.NODE_ENV which is set 'production'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';

import store from './stores';

const enhancer = (process.env.NODE_ENV == 'development')
               ? composeWithDevTools(applyMiddleware(thunk))
               : applyMiddleware(thunk);

export default () => createStore(store, enhancer);
