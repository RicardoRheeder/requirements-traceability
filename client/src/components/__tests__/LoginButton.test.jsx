import React from "react"
import Enzyme, { shallow, mount} from "enzyme"
import LoginButton from "../LoginButton"
import Adapter from "enzyme-adapter-react-16"
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"


const mockStore = configureStore([])
Enzyme.configure({adapter: new Adapter() });

describe("Login Button", () => {
    let store;
    let testButton;
    beforeEach(() => {
        store = mockStore({});
    });
    test("renders", () => {
        const wrapper = shallow(<Provider store = {store}><LoginButton/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
    test("renders with children", () => {
        const wrapper = mount(<Provider store = {store}><LoginButton/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});