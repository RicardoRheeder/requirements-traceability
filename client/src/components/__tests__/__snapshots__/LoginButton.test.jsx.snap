// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Button renders 1`] = `ShallowWrapper {}`;
